OMGVA PoS/Business layer/Controllers/BusinessController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using OMGVA_PoS.Data_layer.DTOs;
+using OMGVA_PoS.Data_layer.Enums;
+using OMGVA_PoS.Data_layer.Models;
+using OMGVA_PoS.Data_layer.Repositories.Business_Management;
+using OMGVA_PoS.Helper_modules.Utilities;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace OMGVA_PoS.Business_layer.Controllers
+{
+    [Route("business")]
+    [ApiController]
+    public class BusinessController(IBusinessRepository businessRepository) : Controller
+    {
+        private readonly IBusinessRepository _businessRepository = businessRepository;
+
+        [HttpGet]
+        [Authorize(Roles = "Admin")]
+        [ProducesResponseType<List<Business>>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        public IActionResult GetAllBusinesses() {
+            return Ok(JsonConvert.SerializeObject(_businessRepository.GetBusinesses()));
+        }
+
+
+        [HttpGet("{id}")]
+        [Authorize(Roles = "Admin,Owner")]
+        [ProducesResponseType<Business>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public IActionResult GetBusiness(long id) {
+            JwtSecurityToken token = JwtTokenHelper.GetJwtToken(HttpContext.Request.Headers.Authorization);
+            if (token.UserRoleEquals(UserRole.Owner) && !token.UserBusinessEquals(id)) {
```

Comment:
Is this condition correct? Seems like `employee` or `admin` could access any business, because first condition would fail (because they are not `owner`) and then move to the next statements to retrieve the business